src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -2077,3 +2076,20 @@ instruct vclearArray_reg_reg(iRegL_R29 cnt, iRegP_R28 base, Universe dummy,
 
   ins_pipe(pipe_class_memory);
 %}
+
+// Vector Load Const
+instruct vloadcon(vReg dst, immI0 src) %{
+  match(Set dst (VectorLoadConst src));
+  ins_cost(VEC_COST);
+  format %{ "vloadcon $dst\t# generate iota indices" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    Assembler::SEW sew = Assembler::elemtype_to_sew(bt);
+    __ vsetvli(t0, x0, sew);
```

Comment:
I heard this opcode ( vsetvli) is pretty costly when the params of vector engine gets reconfigured ( for example for different element width). Not saying anything bad here. We might need to think about some optimisations for using vsetvli in future

Reply:
Hi @VladimirKempik , thanks for the review! Almost every instruct in `riscv_v.ad ` uses this opcode ( vsetvli)  at the beginning, and it does look like there is a need for optimization. Maybe we can probably discuss it more extensively and change it uniformly.

Reply:
> Hi @VladimirKempik , thanks for the review! Almost every instruct in `riscv_v.ad ` uses this opcode ( vsetvli) at the beginning, and it does look like there is a need for optimization. Maybe we can probably discuss it more extensively and change it uniformly.

It's interesting to know how native compilers like GCC/LLVM eliminate such redundances. I guess they face similar issues when do auto-vectorization for different loops within the same function.